File: backend/models/questionModel.go
package models

type Question struct {
	ID          string `bson:"_id" json:"_id"`
	Title       string `bson:"title"`
	Description string `bson:"description"`
	Categories  string `bson:"categories"`
	Complexity  string `bson:"complexity"`
	Link        string `bson:"link"`
}

File: backend/models/questionModel.js
const mongoose = require('mongoose')

// enum values for categories and complexities of a question.
// const CATEGORIES = ['Strings', 'Algorithms', 'Data Structures', 'Bit Manipulation', 'Recursion', 'Databases', 'Brainteaser', 'Arrays']
const COMPLEXITIES = ['Easy', 'Medium', 'Hard']

const questionSchema = new mongoose.Schema({
    title: {
        type: String,
        required: true
    }, 
    description: {
        type: String,
        required: true
    },
    category: {
        type: String,
        required: true
    },
    complexity: {
        type: String,
        enum: COMPLEXITIES,
        required: true
    }
});

module.exports = mongoose.model('Question', questionSchema)